// react-dump/lib/App.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _jsxFileName = 'src/App.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('./App.css');

var _Reactdump = require('./components/Reactdump');

var _Reactdump2 = _interopRequireDefault(_Reactdump);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var App = function (_React$Component) {
  _inherits(App, _React$Component);

  function App() {
    _classCallCheck(this, App);

    return _possibleConstructorReturn(this, (App.__proto__ || Object.getPrototypeOf(App)).apply(this, arguments));
  }

  _createClass(App, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      // build object to be dumped
      var validate = function validate() {
        return false;
      };
      var updateSession = function updateSession() {};
      var signIn = function signIn(username, password) {
        // validate username and password
        if (!validate(username, password)) return false;
        updateSession();
        _this2.signedIn = true;
        return true;
      };
      var user = {
        firstName: 'Charles',
        lastName: 'Teague',
        age: 21,
        signedIn: false,
        signIn: signIn,
        projects: [{
          name: 'Allaire Spectra',
          status: 'Horrible death'
        }, {
          name: 'ColdFusion 4.5',
          status: 'Been there done that'
        }]
      };

      // <Dump obj={user} label="top-one" />
      return _react2.default.createElement(
        'div',
        { className: 'App', __source: {
            fileName: _jsxFileName,
            lineNumber: 37
          },
          __self: this
        },
        _react2.default.createElement(
          'div',
          { className: 'App-header', __source: {
              fileName: _jsxFileName,
              lineNumber: 38
            },
            __self: this
          },
          _react2.default.createElement(
            'h2',
            {
              __source: {
                fileName: _jsxFileName,
                lineNumber: 39
              },
              __self: this
            },
            'Welcome to React'
          )
        ),
        _react2.default.createElement(_Reactdump2.default, { obj: user, label: 'top-one', expand: 'true', __source: {
            fileName: _jsxFileName,
            lineNumber: 41
          },
          __self: this
        })
      );
    }
  }]);

  return App;
}(_react2.default.Component);

exports.default = App;
