// react-dump/lib/components/Arr.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _jsxFileName = 'src/components/Arr.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _DataTyper = require('./DataTyper');

var _DataTyper2 = _interopRequireDefault(_DataTyper);

var _Table = require('./Table');

var _Table2 = _interopRequireDefault(_Table);

var _Row = require('./Row');

var _Row2 = _interopRequireDefault(_Row);

var _RowHeader = require('./RowHeader');

var _RowHeader2 = _interopRequireDefault(_RowHeader);

var _CircularReference = require('./CircularReference');

var _CircularReference2 = _interopRequireDefault(_CircularReference);

var _getPathToCircularRef = require('./getPathToCircularRef');

var _getPathToCircularRef2 = _interopRequireDefault(_getPathToCircularRef);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// assigned to component after component definition, no hoisting within
var defaultProps = {
  obj: [],
  opts: {
    label: 'Array',
    expand: true
  },
  cache: {
    bFilteredLevel: false,
    level: 0,
    objects: [],
    paths: []
  },
  currentPath: []
};

var Arr = function (_React$Component) {
  _inherits(Arr, _React$Component);

  function Arr() {
    _classCallCheck(this, Arr);

    return _possibleConstructorReturn(this, (Arr.__proto__ || Object.getPrototypeOf(Arr)).apply(this, arguments));
  }

  _createClass(Arr, [{
    key: 'render',
    value: function render() {
      var obj = this.props.obj;
      var sparseKeys = Object.keys(obj);
      var cache = this.props.cache;
      var currentPath = this.props.currentPath;
      var rows = '[empty]'; // assume empty
      if (sparseKeys.length) {
        rows = [];
      }
      var label = this.props.opts.label + ' - (' + sparseKeys.length + ')';

      // if a circular ref, write it and be done
      var circPath = (0, _getPathToCircularRef2.default)(obj, cache, currentPath);
      if (circPath.length > 0) {
        return _react2.default.createElement(_CircularReference2.default, { expand: this.props.opts.expand, circPath: circPath, __source: {
            fileName: _jsxFileName,
            lineNumber: 42
          },
          __self: this
        });
      }

      cache.level++;

      var _iteratorNormalCompletion = true;
      var _didIteratorError = false;
      var _iteratorError = undefined;

      try {
        for (var _iterator = sparseKeys[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
          var i = _step.value;

          var element = obj[i];
          var subPath = [].concat(_toConsumableArray(currentPath));
          subPath.push(i);
          rows.push(_react2.default.createElement(
            _Row2.default,
            { key: i, className: 'reactdump-label reactdump-Array', label: i, expand: this.props.opts.expand, __source: {
                fileName: _jsxFileName,
                lineNumber: 51
              },
              __self: this
            },
            _react2.default.createElement(_DataTyper2.default, { obj: element, opts: this.props.opts, cache: cache, currentPath: subPath, __source: {
                fileName: _jsxFileName,
                lineNumber: 51
              },
              __self: this
            })
          ));
        }
      } catch (err) {
        _didIteratorError = true;
        _iteratorError = err;
      } finally {
        try {
          if (!_iteratorNormalCompletion && _iterator.return) {
            _iterator.return();
          }
        } finally {
          if (_didIteratorError) {
            throw _iteratorError;
          }
        }
      }

      return _react2.default.createElement(
        _Table2.default,
        { className: 'reactdump reactdump-Array', __source: {
            fileName: _jsxFileName,
            lineNumber: 55
          },
          __self: this
        },
        _react2.default.createElement(_RowHeader2.default, { className: 'reactdump-label reactdump-Array', label: label, expand: this.props.opts.expand, __source: {
            fileName: _jsxFileName,
            lineNumber: 56
          },
          __self: this
        }),
        rows
      );
    }
  }]);

  return Arr;
}(_react2.default.Component);

exports.default = Arr;

Arr.defaultProps = defaultProps;
