// react-dump/lib/components/Boolean.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _jsxFileName = 'src/components/Boolean.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Table = require('./Table');

var _Table2 = _interopRequireDefault(_Table);

var _Row = require('./Row');

var _Row2 = _interopRequireDefault(_Row);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// assigned to component after component definition, no hoisting within
var defaultProps = {
  obj: null,
  opts: {
    label: 'Boolean',
    expand: true
  }
};

var Boolean = function (_React$Component) {
  _inherits(Boolean, _React$Component);

  function Boolean() {
    _classCallCheck(this, Boolean);

    return _possibleConstructorReturn(this, (Boolean.__proto__ || Object.getPrototypeOf(Boolean)).apply(this, arguments));
  }

  _createClass(Boolean, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Table2.default,
        { className: 'reactdump reactdump-Boolean', __source: {
            fileName: _jsxFileName,
            lineNumber: 17
          },
          __self: this
        },
        _react2.default.createElement(
          _Row2.default,
          { className: 'reactdump-label reactdump-Boolean', label: this.props.opts.label, expand: this.props.opts.expand, expandCell: this.props.opts.expand, __source: {
              fileName: _jsxFileName,
              lineNumber: 18
            },
            __self: this
          },
          _react2.default.createElement(
            'span',
            { className: this.props.obj ? 'reactdump-yes' : 'reactdump-no', __source: {
                fileName: _jsxFileName,
                lineNumber: 19
              },
              __self: this
            },
            this.props.obj.toString()
          )
        )
      );
    }
  }]);

  return Boolean;
}(_react2.default.Component);

exports.default = Boolean;


Boolean.defaultProps = defaultProps;
