// react-dump/lib/components/DataTyper.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _jsxFileName = 'src/components/DataTyper.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _getObjectClassName = require('./getObjectClassName');

var _getObjectClassName2 = _interopRequireDefault(_getObjectClassName);

var _Boolean = require('./Boolean');

var _Boolean2 = _interopRequireDefault(_Boolean);

var _String = require('./String');

var _String2 = _interopRequireDefault(_String);

var _Math = require('./Math');

var _Math2 = _interopRequireDefault(_Math);

var _Undefined = require('./Undefined');

var _Undefined2 = _interopRequireDefault(_Undefined);

var _Null = require('./Null');

var _Null2 = _interopRequireDefault(_Null);

var _Date = require('./Date');

var _Date2 = _interopRequireDefault(_Date);

var _Number = require('./Number');

var _Number2 = _interopRequireDefault(_Number);

var _RegExp = require('./RegExp');

var _RegExp2 = _interopRequireDefault(_RegExp);

var _Error = require('./Error');

var _Error2 = _interopRequireDefault(_Error);

var _Function = require('./Function');

var _Function2 = _interopRequireDefault(_Function);

var _Arr = require('./Arr');

var _Arr2 = _interopRequireDefault(_Arr);

var _Obj = require('./Obj');

var _Obj2 = _interopRequireDefault(_Obj);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// assigned to component after component definition, no hoisting within
var defaultProps = {
  obj: null,
  opts: {
    label: '' // string header for top level header dump output
    , expand: true // boolean, expands views
  },
  cache: {
    bFilteredLevel: false,
    level: 0,
    objects: [],
    paths: []
  },
  currentPath: []
};

/*
// in previous versions, expand could be an array of object class names to expand
, opts.output: 'browser' // NOT IMPLEMENTED where to send results browser|console|file
, opts.format: 'html' // NOT IMPLEMENTED output text or HTML format
, opts.hide: null // NOT IMPLEMENTED hide column or keys
, opts.keys: null // NOT IMPLEMENTED For a structure, number of keys to display
, opts.show: null // NOT IMPLEMENTED show column or keys
, opts.top: null // NOT IMPLEMENTED The number of rows to display. For a structure, this is the number of nested levels to display (useful for large stuctures)
*/

var DataTyper = function (_React$Component) {
  _inherits(DataTyper, _React$Component);

  function DataTyper() {
    _classCallCheck(this, DataTyper);

    return _possibleConstructorReturn(this, (DataTyper.__proto__ || Object.getPrototypeOf(DataTyper)).apply(this, arguments));
  }

  _createClass(DataTyper, [{
    key: 'render',
    value: function render() {
      var objectClassName = (0, _getObjectClassName2.default)(this.props.obj);
      var opts = Object.assign({}, this.props.opts);
      opts.label = objectClassName;
      if (this.props.currentPath.length === 1) {
        opts.label = this.props.opts.label + ' - ' + objectClassName;
      }

      var classes = {
        Array: _react2.default.createElement(_Arr2.default, { obj: this.props.obj, opts: opts, cache: this.props.cache, currentPath: this.props.currentPath, __source: {
            fileName: _jsxFileName,
            lineNumber: 55
          },
          __self: this
        }),
        Boolean: _react2.default.createElement(_Boolean2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 56
          },
          __self: this
        }),
        Date: _react2.default.createElement(_Date2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 57
          },
          __self: this
        }),
        Error: _react2.default.createElement(_Error2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 58
          },
          __self: this
        }),
        Function: _react2.default.createElement(_Function2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 59
          },
          __self: this
        }),
        Math: _react2.default.createElement(_Math2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 60
          },
          __self: this
        }),
        Null: _react2.default.createElement(_Null2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 61
          },
          __self: this
        }),
        Number: _react2.default.createElement(_Number2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 62
          },
          __self: this
        }),
        Object: _react2.default.createElement(_Obj2.default, { obj: this.props.obj, opts: opts, cache: this.props.cache, currentPath: this.props.currentPath, __source: {
            fileName: _jsxFileName,
            lineNumber: 63
          },
          __self: this
        }),
        RegExp: _react2.default.createElement(_RegExp2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 64
          },
          __self: this
        }),
        String: _react2.default.createElement(_String2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 65
          },
          __self: this
        }),
        Undefined: _react2.default.createElement(_Undefined2.default, { obj: this.props.obj, opts: opts, __source: {
            fileName: _jsxFileName,
            lineNumber: 66
          },
          __self: this
        })
      };

      // when found in the classes array:
      if (Object.keys(classes).indexOf(objectClassName) >= 0) {
        return classes[objectClassName];
      }
      return _react2.default.createElement(_Error2.default, { obj: this.props.obj, opts: { label: 'Unknown Class', expand: opts.expand }, __source: {
          fileName: _jsxFileName,
          lineNumber: 73
        },
        __self: this
      });
    }
  }]);

  return DataTyper;
}(_react2.default.Component);

exports.default = DataTyper;


DataTyper.defaultProps = defaultProps;
