// react-dump/lib/components/Function.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _jsxFileName = 'src/components/Function.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Table = require('./Table');

var _Table2 = _interopRequireDefault(_Table);

var _Row = require('./Row');

var _Row2 = _interopRequireDefault(_Row);

var _RowHeader = require('./RowHeader');

var _RowHeader2 = _interopRequireDefault(_RowHeader);

var _escapeHtml = require('./escapeHtml');

var _escapeHtml2 = _interopRequireDefault(_escapeHtml);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// assigned to component after component definition, no hoisting within
var defaultProps = {
  obj: null,
  opts: {
    label: 'Function',
    expand: true
  }
};

var Function = function (_React$Component) {
  _inherits(Function, _React$Component);

  function Function() {
    _classCallCheck(this, Function);

    return _possibleConstructorReturn(this, (Function.__proto__ || Object.getPrototypeOf(Function)).apply(this, arguments));
  }

  _createClass(Function, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Table2.default,
        { className: 'reactdump reactdump-Function', __source: {
            fileName: _jsxFileName,
            lineNumber: 19
          },
          __self: this
        },
        _react2.default.createElement(_RowHeader2.default, { className: 'reactdump-label reactdump-Function', label: this.props.opts.label, expand: this.props.opts.expand, cols: '1', __source: {
            fileName: _jsxFileName,
            lineNumber: 20
          },
          __self: this
        }),
        _react2.default.createElement(
          _Row2.default,
          { className: 'reactdump-label reactdump-Function', label: this.props.opts.label, expand: this.props.opts.expand, cols: '1', __source: {
              fileName: _jsxFileName,
              lineNumber: 21
            },
            __self: this
          },
          _react2.default.createElement(
            'pre',
            {
              __source: {
                fileName: _jsxFileName,
                lineNumber: 22
              },
              __self: this
            },
            _react2.default.createElement(
              'code',
              { className: 'lang-javascript', __source: {
                  fileName: _jsxFileName,
                  lineNumber: 23
                },
                __self: this
              },
              (0, _escapeHtml2.default)(this.props.obj.toString().replace(/\t/g, ''))
            )
          )
        )
      );
    }
  }]);

  return Function;
}(_react2.default.Component);

exports.default = Function;


Function.defaultProps = defaultProps;
