// react-dump/lib/components/RowHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _jsxFileName = 'src/components/RowHeader.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// assigned to component after component definition, no hoisting within
var defaultProps = {
  className: '',
  label: '',
  expand: true,
  title: '',
  cols: 2
};

var RowHeader = function (_React$Component) {
  _inherits(RowHeader, _React$Component);

  function RowHeader() {
    _classCallCheck(this, RowHeader);

    return _possibleConstructorReturn(this, (RowHeader.__proto__ || Object.getPrototypeOf(RowHeader)).apply(this, arguments));
  }

  _createClass(RowHeader, [{
    key: 'render',
    value: function render() {
      switch (this.props.cols) {
        case 0:
          return _react2.default.createElement(
            'tr',
            {
              __source: {
                fileName: _jsxFileName,
                lineNumber: 18
              },
              __self: this
            },
            _react2.default.createElement(
              'th',
              { className: this.props.className, onClick: this.handleClick, __source: {
                  fileName: _jsxFileName,
                  lineNumber: 18
                },
                __self: this
              },
              '[EMPTY]'
            )
          );

        default:
          if (this.props.expand) {
            return _react2.default.createElement(
              'tr',
              {
                __source: {
                  fileName: _jsxFileName,
                  lineNumber: 24
                },
                __self: this
              },
              _react2.default.createElement(
                'th',
                { colSpan: this.props.cols, className: this.props.className, onClick: this.handleClick, __source: {
                    fileName: _jsxFileName,
                    lineNumber: 25
                  },
                  __self: this
                },
                this.props.label
              )
            );
          }
          return _react2.default.createElement(
            'tr',
            { style: this.state.expand ? {} : { display: 'none' }, __source: {
                fileName: _jsxFileName,
                lineNumber: 32
              },
              __self: this
            },
            _react2.default.createElement(
              'th',
              { colSpan: this.props.cols, className: this.props.className, style: { display: 'none' }, onClick: this.handleClick, __source: {
                  fileName: _jsxFileName,
                  lineNumber: 33
                },
                __self: this
              },
              this.props.label
            )
          );
      } // /switch
    }
  }]);

  return RowHeader;
}(_react2.default.Component);

exports.default = RowHeader;


RowHeader.defaultProps = defaultProps;
